doc/vxlan/data_plane_VXLAN_HLD.md

```diff
@@ -0,0 +1,507 @@
+1. Remote VTEP name and IP address configuration.
+2. Remote VTEP-VNI association.
+
+##### Warm Boot Requirements
+Warm reboot is intended to be supported for the following cases:
+
+- Planned system warm reboot.
+- Planned SwSS warm reboot.
+
+
+### Functional description
+
+##### L2 unicast forwarding over VXLAN
+L2 unicast traffic is forwarded to the tunnel to which the MAC points to. Traffic received from VXLAN tunnels are mapped to a VLAN based on the received VNI and the VNI-VLAN lookup. The received traffic can be L2 unicast or BUM depending on the inner MAC lookup.
+![L2 unicast flow](images/uc_l2_vxlan_packet_flow.png)
+
+##### L2 flooding over VXLAN
+Traffic received from the access ports are L2 forwarded over VXLAN tunnels if the DMAC lookup result does not match the switch MAC. The frame format is as described in RFC 7348. The VNI is based on the configured global VLAN-VNI map. BUM traffic is ingress replicated to all the tunnels which are part of the VLAN. The DIP of the BUM packets is the IP address of the remote VTEP. Traffic received from VXLAN tunnels are never forwarded onto another VXLAN tunnels.
+![L2 flooding flow](images/L2_flooding_vxlan_packet_flow.png)
+
+##### Basic Approach
+
+* Per each pair of Local VTEP-Remote VTEP, P2P tunnel is created.
+* Each remote VTEP is represented by bridge port with MAC learning enabled.
+* Per each remote VTEP - tunnel start and termination created.
+
+![Devices connection](images/basic_approach_device_overview.png)
+
+Local traffic will be mapped accoding to VLAN-VNI configuration and send to port which represent correspong remote VTEP.
+
+![Tunnel start traffic](images/basic_approach_tunnel_start.png)
+
+Traffic received on ports, that represent remote VTEPs, will be tunnel terminated, remote MAC will be learned on this remote port and traffic will be mapped according to VLAN-VNI map to local port.
+
+![Tunnel termination traffic](images/basic_approach_tunnel_term.png)
+
+
+### Architecture Design
+
+No new blocks are required.
+
+
+#### Container
+
+No new container is added. The changes are added to existing container SwSS. The details of the changes will be discussed in the Design section below.
+
+
+
+---
+##### Figure 1. Current architecture
+![Current VXLAN ARchitecture](images/curent_vxlan_arch.png)
+---
+
+##### Figure 1. Dataplane architecture
+![Dataplane VXLAN ARchitecture](images\dt_vxlan_arch.png)
+---
+
+#### SAI API
+No changes to SAI APIs are required.
+
+### 2. Configuration and management
+
+#### CLI Enhancements
+
+##### 5.1 Remote VTEP IP configuration
+
+* Ability to add and remove remote VTEP
+   - config vxlan remote add <vtepname> <ipv4-address>
+   - config vxlan remote del <vtepname>
+   - vtepname is a string. 
+   - ipv4-address is an IPV4 address in dotted notation A.B.C.D
+
+##### 5.2 Remote VTEP-VNI association configuration 
+
+* Ability to add and remove remote VTEP assiciation to VNI
+   - config vxlan remote map add <vtepname> <vni-id>
```

Comment:
It might be more intuitive for the user to configure the VLANID instead of VNI.

Reply:
Do you both vlan and vni ? Like:  config vxlan remote map add <vtepname> <vlan-id> <vni-id>